### DISCUSSION TITLE
Auto completion search

### DISCUSSION THREAD

**Discussion by jesussmile** [liodali/osm_flutter#88] [General]
Is there any way I can get the auto-completion feature in the search bar or could you please mention any providers who provide it for free?

**Comment by liodali** [comment]
Hi,
I already creat static method to make auto completion seach for address 
and also i create custom widget for that purpose

**Comment by liodali** [comment]
you can use this method : 
```dart
 List<SearchInfo> suggestions = await addressSuggestion("address");
```

**Comment by liodali** [comment]
I used this  api : https://photon.komoot.io/
you can used if you want custom method

**Reply by jesussmile** [reply]
Apologies, The search took a while to load and thought it only worked after hitting the right arrow at the bottom.  I understand it now. Thank you.

**Reply by liodali** [reply]
no it  only start look for suggestion after  3 caracters 
to minimize the number of calls to the endpoint

**Reply by liodali** [reply]
also that api has rate limit we cannot make lot of calls in 1 second that's why I limit search by 3 caractere
for example if you are looking for London it will start after Lon nott before to be also accurate

**Reply by liodali** [reply]
but you can use `addressSuggestion` directly with you own widget

**Reply by jesussmile** [reply]
> also that API has rate limit we cannot make lot of calls in 1 second that's why I limit search by 3 caractere
> for example if you are looking for London it will start after Lon nott before to be also accurate

Yes I understand, I read your documents.  That's clever. by the way, how is the ios plugin coming on ? with android I am getting extremely good results with your plugin. Thank you for the hard work.

**Reply by liodali** [reply]
I'm suggling with showing a road in the map , i have 2 week in this bug, i don't know why and the problem that i can't test it in my iphone  because i'm using hackintosh. i will try another solution maybe i will use webview to show map in ios side because all resource in ios are paid except mapkit

**Reply by liodali** [reply]
or i will create my own map tiles provider and used with mapbox sdk combined with osm api 
but this solution will take lot of time, i will go with it after i make small support for the ios side

**Reply by jesussmile** [reply]
I am sure you will overcome this.  Take your time no worries.  Thank you for your hard work.

**Reply by liodali** [reply]
thanks, i try to do my best.
i will try at least in next month i publish version that support ios  a minimum of feature ( like draw road,select poition) 
if you know someone that good in ios dev and he want to contribute or help , you can invite him

**Reply by jesussmile** [reply]
what you are doing is pretty big,  my circle isn't up to your standards. perhaps you could post this in a bigger forum with more experienced people? I don't think enough people know about this.
